app/controllers/qunit_controller.rb

```diff
@@ -6,4 +6,10 @@ class QunitController < ApplicationController
   def index
     raise Discourse::InvalidAccess.new if Rails.env.production?
   end
+
+  # make nonce static and set restrictive csp for testing
+  def set_csp_header
+    request.env["nonce"] = "1234"
+    response.headers["Content-Security-Policy"] = "script-src 'nonce-1234' 'unsafe-eval';"
```

Comment:
Use a restrictive CSP in qunit to ensure that nonce-ing is working properly for all (tested) `<script>`s